[DOCS] add Local Python Install to docs

petermoyer opened the pull request:
Also add 3 new components

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Add three new components to install
- Add local python install
- Add Spark EMR template
- Update sidebar.js (fixes typo and adds emoji)

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

### Definition of Done
Please delete options that are not relevant.

- [ X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [X ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ X] I have run any local integration tests and made sure that nothing is broken.


Thank you for submitting!

petermoyer committed:
Gdoc-26 add Local Python Install to docs

Also add 3 new components

petermoyer commented:
> Also add 3 new components
> 
> Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).
> 
> Changes proposed in this pull request:
> 
> * Add three new components to install
> * Add local python install
> * Add Spark EMR template
> * Update sidebar.js (fixes typo and adds emoji)
> 
> After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.
> 
> ### Definition of Done
> Please delete options that are not relevant.
> 
> * [ X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
> * [X ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
> * [ X] I have commented my code, particularly in hard-to-understand areas
> * [ X] I have made corresponding changes to the documentation
> * [ ]  I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
> * [ X] I have run any local integration tests and made sure that nothing is broken.
> 
> Thank you for submitting!

petermoyer commented:
@cla-bot check

Commit:
Merge branch 'develop' into DOCS/GDOC-107/GDOC-26/local_ge_install

Aylr commented on docs/guides/setup/installing/local.md:
```diff
@@ -1,5 +1,89 @@
 ---
-title: local python
+title: Installing in great_expectations in Local Python3
```
Please make titles consistent with the specific language in the design document: https://github.com/superconductive/design/blob/main/docs/20210421_knoxpod_golden_paths_mental_map/toc.md#installation-this-still-needs-to-be-fleshed-out-and-finalized

petermoyer replied:
Done

Aylr commented on docs/guides/setup/installing/spark-emr.md:
```diff
@@ -1,6 +1,48 @@
 ---
-title: Installing great_expectations in Spark EMR 
+title: Installing great_expectations in Spark EMR cluster
```
Please make titles consistent with the specific language in the design document: https://github.com/superconductive/design/blob/main/docs/20210421_knoxpod_golden_paths_mental_map/toc.md#installation-this-still-needs-to-be-fleshed-out-and-finalized

petermoyer replied:
changed

Aylr commented on docs/guides/setup/components/install_prereq.jsx:
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
```
Rather than duplicating this component, perhaps we can move the existing one out to a more common place so it can be shared. We may want to factor out `defaultPrerequisiteItems` so it is more general and reusable. Happy to pair if you'd like!

petermoyer replied:
Will be handled in the next PR as it requires more changes

Aylr commented on sidebars.js:
```diff
@@ -32,7 +32,7 @@ module.exports = {
             {
               type: 'category',
-              label: 'Installing',
+              label: '🧰 Installing',
```
`Installation` per the design spec. Thanks!

Aylr replied:
Fixed

Aylr reviewed (changes requested):
Awesome work. A few change requests.

Shinnnyshinshin commented on docs/guides/setup/installing/local.md:
````diff
@@ -1,5 +1,89 @@
+
+### 2. Install Great Expectations
+```console
+pip install great_expectations
````
Regarding installation, running `great_expectatsions --v3-api init` seems out of scope (and would be handled by a different doc). 

would we be able to have Step 2 be the following 

```
pip install great_expectations

great_expectations --version
```

And then have a link to 

`docs/guides/setup/configuring-data-contexts/how-to-create-a-new-data-context-with-the-cli.md?`

- this guide should cover the init and walking through the CLI steps.

petermoyer replied:
Done

Shinnnyshinshin commented on docs/guides/setup/installing/local.md:
```diff
@@ -1,5 +1,89 @@
+
+## Additional Notes
+
+To view the full scripts used in this page, see them on GitHub:
```
This can be removed

Shinnnyshinshin commented on docs/guides/setup/installing/spark-emr.md:
````diff
@@ -1,6 +1,48 @@
+
+### 1. Install Great Expectations
+
+The guide demonstrates the recommended path for instantiating a Data Context without a full configuration directory and without using the [Great Expectations Command Line Interface (CLI)](/docs/guides/setup/configuring-data-contexts/how-to-create-a-new-data-context-with-the-cli)
+
+```console
+sc.install_pypi_package("great_expectations")
+```
+
+### 2. Configure a Data Context in code
+Follow the steps for creating an in-code Data Context in [How to instantiate a Data Context without a yml file](/docs/guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file)
+
+The snippet at the end of the guide shows Python code that instantiates and configures a Data Context in code for an EMR Spark cluster. Copy this snippet into a cell in your EMR Spark notebook or use the other examples to customize your configuration.
+
+Test your configuration.
+
+Execute the cell with the snippet above.
+
+Then copy this code snippet into a cell in your EMR Spark notebook, run it and verify that no error is displayed:
+```console
+context.list_datasources()
+```
+
+<Congratulations />
+
+## Additional Notes
+
+To view the full scripts used in this page, see them on GitHub:
+
+## Next Steps
+
+<NextSteps />
````
I know that this was ported over from our previous docs, but in the way that we are writing the new docs, I'd like to request the following changes: 

Step 1:
- please include some additional information on where this command should be run:
`sc.install_pypi_package("great_expectations")`
-  Is it in a Notebook? is there a separate console?
- Also, to have parity with the local installation guide (`local.md`), it would be nice to have a command like `great_expectations --version` (or an EMR equivalent) to checking whether `great_expectations` was installed correctly. 

`Step 2` would become `Next Steps` and would say something like: 

In an EMR Spark Notebook, you will not have direct access to the CLI. Therefore you will need to create a Data Context in memory. To find out how please look at the document on  [How to instantiate a Data Context without a yml file](/docs/guides/setup/configuring-data-contexts/how-to-instantiate-a-data-context-without-a-yml-file)

Shinnnyshinshin replied:
The link will eventually move to a `Deployment Patterns` document called `How to instantiate a Data Context on an EMR Spark cluster` but it's not in the current branch (is being included by #2915)

petermoyer replied:
Will be handled in a future PR

Shinnnyshinshin commented on sidebars.js:
```diff
@@ -281,7 +281,7 @@ module.exports = {
         {
           type: 'category',
-          label: 'Contributingt',
+          label: 'Contributing',
           items: [
```
Thank you for catching this :)

Shinnnyshinshin reviewed (dismissed):
Thank you very very much for these guides.  A few changes from me as well

petermoyer committed:
Updates on the comments

Fix syntax for titles. Remove extraneuous text.
Other comments will be handled in next PR with spliiting out the jsx files

petermoyer committed:
Merge branch 'develop' into DOCS/GDOC-107/GDOC-26/local_ge_install

petermoyer committed:
Merge branch 'develop' into DOCS/GDOC-107/GDOC-26/local_ge_install

petermoyer committed:
Merge branch 'develop' into DOCS/GDOC-107/GDOC-26/local_ge_install

Aylr committed:
Merge branch 'develop' into DOCS/GDOC-107/GDOC-26/local_ge_install

Aylr committed:
added missing requirements file to docs CI pipeline
